homeassistant/components/sunweg/config_flow.py

```diff
@@ -72,3 +98,37 @@ async def async_step_plant(self, user_input=None) -> ConfigFlowResult:
         self._abort_if_unique_id_configured()
         self.data.update(user_input)
         return self.async_create_entry(title=self.data[CONF_NAME], data=self.data)
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle reauthorization request from SunWEG."""
+        self.data.update(entry_data)
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle reauthorization flow."""
+        if user_input is None:
+            return self._async_show_user_form("reauth_confirm")
+
+        self.data.update(user_input)
+        conf_result = await self.hass.async_add_executor_job(
+            self._set_auth_data,
+            "reauth_confirm",
+            user_input[CONF_USERNAME],
+            user_input[CONF_PASSWORD],
+        )
+        if conf_result is not None:
+            return conf_result
+
+        entry = await self.async_set_unique_id(self.context[CONF_UNIQUE_ID])
```

Comment:
Why do you get the config entry indirectly via the unique id instead of directly via the entry id?
Also, `CONF_UNIQUE_ID` is not really correct here, just use a hard coded string instead.